### ISSUE TITLE
If use “freeze layer” on base_model(resnet50 notop) to training,  problem in validation accuracy

### ISSUE THREAD

**Issue by taobiaoli** [keras-team/keras#12317]
### When I use the freezing layer, the training accuracy is normal, but the verification accuracy is wrong, sometimes 0, sometimes 1, or other values, in short, the jump is larger.And the test accuracy  is 0.5 (two classifications)

Epoch 1/10
 - 49s - loss: 0.1015 - acc: 0.9621 - val_loss: 0.7007 - val_acc: 0.3543
Epoch 2/10
 - 45s - loss: 0.0647 - acc: 0.9758 - val_loss: 0.5447 - val_acc: 0.9992
Epoch 3/10
 - 44s - loss: 0.0561 - acc: 0.9784 - val_loss: 0.5452 - val_acc: 0.9992
Epoch 4/10
 - 46s - loss: 0.0510 - acc: 0.9802 - val_loss: 0.4455 - val_acc: 1.0000
Epoch 5/10
 - 45s - loss: 0.0491 - acc: 0.9812 - val_loss: 0.5055 - val_acc: 0.9998
Epoch 6/10
 - 45s - loss: 0.0457 - acc: 0.9829 - val_loss: 0.4265 - val_acc: 1.0000
Epoch 7/10
 - 45s - loss: 0.0440 - acc: 0.9838 - val_loss: 0.3548 - val_acc: 1.0000
Epoch 8/10
 - 44s - loss: 0.0444 - acc: 0.9832 - val_loss: 0.3011 - val_acc: 1.0000
Epoch 9/10
 - 45s - loss: 0.0426 - acc: 0.9838 - val_loss: 0.3676 - val_acc: 1.0000
Epoch 10/10
 - 45s - loss: 0.0419 - acc: 0.9846 - val_loss: 0.3014 - val_acc: 1.0000
5000/5000 [==============================] - 12s 2ms/step
test loss; 0.8240634125709534
test accuracy: 0.5

### As the code shows, I just commented out two lines of code.
    #for layer in base_model.layers:
    #    layer.trainable = False
Train on 16000 samples, validate on 4000 samples
Epoch 1/10
 - 116s - loss: 0.0679 - acc: 0.9733 - val_loss: 0.0389 - val_acc: 0.9848
Epoch 2/10
 - 105s - loss: 0.0145 - acc: 0.9956 - val_loss: 0.0662 - val_acc: 0.9725
Epoch 3/10
 - 105s - loss: 0.0078 - acc: 0.9978 - val_loss: 0.0470 - val_acc: 0.9822
Epoch 4/10
 - 105s - loss: 0.0043 - acc: 0.9989 - val_loss: 0.0259 - val_acc: 0.9910
Epoch 5/10
 - 105s - loss: 0.0033 - acc: 0.9992 - val_loss: 0.0310 - val_acc: 0.9902
Epoch 6/10
 - 104s - loss: 0.0025 - acc: 0.9995 - val_loss: 0.0385 - val_acc: 0.9872
Epoch 7/10
 - 104s - loss: 0.0023 - acc: 0.9995 - val_loss: 0.0423 - val_acc: 0.9870
Epoch 8/10
 - 104s - loss: 0.0016 - acc: 0.9998 - val_loss: 0.0453 - val_acc: 0.9842
Epoch 9/10
 - 105s - loss: 9.6570e-04 - acc: 0.9998 - val_loss: 0.0416 - val_acc: 0.9875
Epoch 10/10
 - 105s - loss: 7.8896e-04 - acc: 0.9999 - val_loss: 0.0410 - val_acc: 0.9875
5000/5000 [==============================] - 12s 2ms/step
test loss; 0.032329649113561024
test accuracy: 0.989

**My code:**

if __name__=="__main__":

    train_data, train_label, test_data, test_label = load_data()
    train_data, test_data = train_data.astype('float32'), test_data.astype('float32')
    train_data, test_data = train_data/255, test_data/255
    train_label = keras.utils.to_categorical(train_label, 2)
    test_label = keras.utils.to_categorical(test_label, 2)
    file_path='cat_dogs_model_resnet50.h5'

    base_model = ResNet50(weights = None, include_top = False, pooling = 'avg')# if use pooling= 'avg' ,later needn't use  GlobalAveragePooling2D.
    #base_model = ResNet50(weights = None, include_top = False)
   
    base_model.load_weights('resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5')
    #x = base_model.output
    #x = GlobalAveragePooling2D()(x)
    #predictions = Dense(2, activation='softmax')(x)
    predictions = Dense(2, activation='softmax')(base_model.output)
    model = Model(inputs=base_model.input, outputs=predictions)
    #for layer in base_model.layers:
    #    layer.trainable = False
    model.compile(optimizer='SGD', 
            loss='binary_crossentropy',  
            metrics=['accuracy'])
    model.summary()
    a = model.get_weights()
    print('training weights before: ',a[-3])
    model.fit(train_data, train_label,epochs=10,batch_size=32,verbose=2,validation_split=0.2,shuffle = True)
    b = model.get_weights()
    print('training weights after: ',b[-3])
    loss, accuracy = model.evaluate(test_data,test_label)
 
    print('test loss;', loss)
    print('test accuracy:', accuracy)
    print('Model Saved.')
    model.save(file_path)

**Comment by sakvaua** [comment]
Try unfreezing all batchnorm layers in your Resnet50, while keeping convolutional frozen.
